crud_for_code_review.py

```diff
@@ -0,0 +1,546 @@
+#     """return list of artists that have multiple genres"""
+
+#     artist_genres = count_genres_by_user_artists(user_id)
+
+#     repeating_artists = []
+#     for artist in artist_genres:
+#         if artist_genres[artist] > 1:
+#             repeating_artists.append(artist)
+
+#     return repeating_artists
+
+
+def optimize_genres(user_id, api_type):
+    """Returns a list of artists by genre with artists in their genre with the 
+    highest count of artists by user
+
+    >>>optimize_genres("test")
+    >>>{'shamanic': ['Beautiful Chorus', 'Rising Appalachia', 'Ayla Nereo'], 
+    'brain waves': ['Alpha Brain Waves'], 'electropop': ['Grimes', 'Sylvan Esso', 
+    'Overcoats', 'Purity Ring', 'Charlotte Day Wilson'],...}"""
+
+    # user_join = User.query.options( 
+    #          db.joinedload('artists') # attribute for user 
+    #            .joinedload('genres')  # attribute from artist
+    #         ).get(user_id)  
+
+    filtered_artists = db.session.query(Artist).filter((UserArtist.api_type==api_type), 
+                    (UserArtist.user_id==user_id), 
+                    (UserArtist.artist_id==Artist.artist_id)).all()
+
+    #count of artists in each genre
+    """{'chillwave': 3, 'dance pop': 3, 'electropop': 6, 
+    'escape room': 4,....}"""
+    artist_genres = count_user_artists_by_genre(user_id) 
+
+    final_dict = {}
+
+    for artist in filtered_artists:
+        max_genre = "" #genre name
+        genre_count = 0 #count of artists in that genre
+        for genre in artist.genres:
+            if artist_genres[genre.genre] == 0:
+                final_dict["No Genre"] = final_dict.get("No Genre", []) + [{"name":artist.artist_name, "value": artist.popularity*100}]
+            #iterates through each artist's genres to find genre with highest
+            #number of associated artists
+            elif artist_genres[genre.genre] >= genre_count:
+                genre_count = artist_genres[genre.genre]
+                max_genre = genre.genre
+        
+        final_dict[max_genre] = final_dict.get(max_genre, []) + [{"name": artist.artist_name, "value": artist.popularity*100}]
+
+    return final_dict
+
+
+def circle_pack_json(user_id, api_type):
+    """Returns formatted input for D3 Circle Pack"""
+    # user_join = User.query.options( 
+    #          db.joinedload('artists') # attribute for user 
+    #            .joinedload('genres')  # attribute from artist
+    #         ).get(user_id)  
+
+    #count of artists in each genre
+    """{'chillwave': 3, 'dance pop': 3, 'electropop': 6, 
+    'escape room': 4,....}"""
+    artist_genres = optimize_genres(user_id, api_type) 
+
+
+    #     >>>{'shamanic': ['Beautiful Chorus', 'Rising Appalachia', 'Ayla Nereo'], 
+    # 'brain waves': ['Alpha Brain Waves'], 'electropop': ['Grimes', 'Sylvan Esso', 
+    # 'Overcoats', 'Purity Ring', 'Charlotte Day Wilson'],...}"""
+
+    data = {"name" : "genres", "children": [] }
+
+    for genre, artists in artist_genres.items():
```

Comment:
nit: `num_artists` instead of `artists`, since you use `artists` elsewhere to describe the whole artist row